src/core/connection.c

```diff
@@ -5511,6 +5511,9 @@ QuicConnRecvDatagramBatch(
         CXPLAT_DBG_ASSERT(Packets[i]->Allocated);
         CXPLAT_ECN_TYPE ECN = CXPLAT_ECN_FROM_TOS(Packets[i]->TypeOfService);
         Packet = Packets[i];
+
+        Connection->Stats.Handshake.HopLimitTTL = Packet->HopLimitTTL; // TODO: Right now, this will update on EVERY packet. How can we filter to just the handshake packets?
```

Comment:
The simplest way to only set it once is only set it if it's zero. Also, instead of being put here, I'd put this on line 5639, right after
```c
Connection->Stats.Recv.TotalBytes += Packet->BufferLength;
```